examples/encfs/README.md

```diff
@@ -67,21 +67,32 @@ az account get-access-token --resource https://managedhsm.azure.net
 
 #### 5. Fill in Key Information
 
-After setting up an [Azure Key Vault resource](#import-key), fill in the `importkeyconfig.json` file with the name of the key to be created and imported into the key vault [Key name](importkeyconfig.json#L3). Additionally, fill in `encfs-sidecar-args.json` with the name of the key to be created and imported into the key vault [Key name](encfs-sidecar-args.json#L9).
+After setting up an Azure Key Vault resource, fill in the `importkeyconfig.json` file with the name of the key to be created and imported into the key vault [Key name](importkeyconfig.json#L3). Additionally, fill in `encfs-sidecar-args.json` with the name of the key to be created and imported into the key vault [Key name](encfs-sidecar-args.json#L9).
 
 Additionally, fill in the optional [key derivation](importkeyconfig.json#L14) for RSA keys and [Key type: `RSA-HSM` or `oct-HSM`](importkeyconfig.json#L4) fields or remove these fields from the `importkeyconfig.json` file. Fill in [key derivation](encfs-sidecar-args.json#L18) and [Key type: `RSA-HSM` or `oct-HSM`](encfs-sidecar-args.json#L10) in `encfs-sidecar-args.json` as well.
 
 
 #### 6. Encrypted Filesystem
 The user needs to instantiate an [Azure Storage Container](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-create?toc=%2Fazure%2Fstorage%2Fblobs%2Ftoc.json&bc=%2Fazure%2Fstorage%2Fblobs%2Fbreadcrumb%2Ftoc.json&tabs=azure-portal) onto which the encrypted filesystem will be uploaded. The roles *Reader* and *Storage Blob Reader* roles need to be assigned to the user-assigned managed identity. Additionally, the role of *Storage Blob Contributor* needs to be assigned for a read-write filesystem.
 
-The script `generatefs/generatefs.sh` creates `encfs.img` with the contents of the `generatefs/filesystem` directory. You may need to adjust the size of the image in the script, as it isn't calculated automatically. 
+The script `generatefs/generatefs.sh` does the following in order: 
 
-The script expects a symmetric key stored in binary format `keyfile.bin` previously created during key import phase. If not passed, the script will generate a new one and the user will need to follow the import key instructions on the new key.
+1. Generates a binary key file named `keyfile.bin` locally if such a file does not exist(see NOTE). 
+2. Creates `encfs.img` with the contents of the `generatefs/filesystem` directory.
+3. Encrypt the file system with `keyfile.bin` 
+
+NOTE: Users have the option of generating `keyfile.bin` on their own by issuing the following command: 
+
+```
+dd if=/dev/random of="<key-file-path>" count=1 bs=32
+```
+Or re-use a pre-existing `keyfile.bin`. In both cases, the script checks the existence of `keyfile.bin` and encrypts the `encfs.img` with it if such a file exists. 
+
+You may need to adjust the size of the image in the script, as it isn't calculated automatically.

```

Comment:
Link it to the line in the script: https://github.com/microsoft/confidential-sidecar-containers/blob/2c76c71ab6610dbabb601ac0c1c9986df1e84151/examples/encfs/generatefs/generatefs.sh#L26